### PR TITLE
mailx: add posibility to configure email's content

### PR DESCRIPTION
**Pull request by mhabrnal**
If formatting file is defined, the reporter-mailx
uses problem report api to create content of emails. Section summary is used
for email's subject. If no formatting file is defined, the standart way for
creating email's content is used.

Related to rhbz#1281312

Signed-off-by: Matej Habrnal <mhabrnal@redhat.com>

### CODE REVIEW

**Comment by mhabrnal** [comment]
Related test: https://github.com/abrt/abrt/pull/1125

**Comment by jfilak** [comment]
How does problem data attaching works in reporter-mailx? I don't see anything related to attachments in the commit.

**Comment by mhabrnal** [comment]
There was only possibility to attach binary files to the email, if ``SendBinaryData`` is ``True``. I've added a possibility to attach elements by using formatting conf file -> all files defined in section ``%attach`` are sent as attachment.

**Comment by jfilak** [comment]
Can we please stop using ``make_description_mailx``? Define a default formatting string instead.

File: src/plugins/reporter-mailx.c [hunk 1]
```diff
@@ -18,6 +18,25 @@
...
+#include "problem_report.h"
+
+#define PR_DEFAULT_SUBJECT \
+    "[abrt] %pkg_name%[[: %crash_function%()]][[: %reason%]][[: TAINTED %tainted_short%]]"
+
+#define PR_MAILX_TEMPLATE \
+    "%%summary:: %s\n" \
+    "\n" \
+    "::" \
+    FILENAME_EXPLOITABLE","FILENAME_NOT_REPORTABLE","FILENAME_REASON"," \
```
Comment (jfilak) [inline src/plugins/reporter-mailx.c]: Please, remove `FILENAME_EXPLOITABLE,FILENAME_NOT_REPORTABLE`.

File: src/plugins/reporter-mailx.c [hunk 2]
```diff
@@ -99,22 +117,52 @@ static void create_and_send_email(
...
+        const char *subject = (env ? env : get_map_string_item_or_NULL(settings, "Subject") ? : PR_DEFAULT_SUBJECT);
+
+        char *format_string = xasprintf(PR_MAILX_TEMPLATE, subject);
+
+        /* attaching binary file to the email */
+        if (send_binary_data)
+            format_string = append_to_malloced_string(format_string, PR_ATTACH_BINARY);
+
+        if (problem_formatter_load_string(pf, format_string))
+            error_msg_and_die("Invalid problem report format string");
```
Comment (jfilak) [inline src/plugins/reporter-mailx.c]: I would explicitly mention which string is invalid ("defautl") and also that it is a bug in code.

File: src/plugins/reporter-mailx.c [hunk 3]
```diff
@@ -99,22 +117,52 @@ static void create_and_send_email(
...
-                args = append_str_to_vector(args, &arg_size, value->content);
-            }
-        }
+        if (problem_formatter_load_file(pf, fmt_file))
+            error_msg_and_die("Invalid format file: %s", fmt_file);
+    }
+
+    problem_report_t *pr = NULL;
+    if (problem_formatter_generate_report(pf, problem_data, &pr))
+        error_msg_and_die("Failed to format bug report from problem data");
```
Comment (jfilak) [inline src/plugins/reporter-mailx.c]: Do we inform users about the used format file in other reporters?

Comment (mhabrnal) [inline src/plugins/reporter-mailx.c]: I do it the same way as it's done in reporter-bugzilla and reporter-rhtsupport.

**Comment by mhabrnal** [comment]
PR updated.

**Comment by jfilak** [comment]
Great job. Thank you!

**Pull request merged**